wled00/bus_manager.cpp

```diff
@@ -605,6 +605,198 @@ void BusNetwork::cleanup() {
   freeData();
 }
 
+// ***************************************************************************
+
+#ifdef WLED_ENABLE_HUB75MATRIX
+
+BusHub75Matrix::BusHub75Matrix(BusConfig &bc) : Bus(bc.type, bc.start, bc.autoWhite) {
+
+  mxconfig.double_buff = false; // <------------- Turn on double buffer
+
+  switch(bc.type) {
+    case 101:
+      mxconfig.mx_width = 32;
+      mxconfig.mx_height = 32;
+      break;
+    case 102:
+      mxconfig.mx_width = 64;
+      mxconfig.mx_height = 32;
+      break;
+    case 103:
+      mxconfig.mx_width = 64;
+      mxconfig.mx_height = 64;
+      break;
+  }
+
+  mxconfig.chain_length = max((u_int8_t) 1, min(bc.pins[0], (u_int8_t) 4)); // prevent bad data preventing boot due to low memory
+
+  if(mxconfig.mx_width >= 64 && (bc.pins[0] > 1)) {
+    DEBUG_PRINTF("WARNING, only single panel can be used of 64 pixel boards due to memory")
+    mxconfig.chain_length = 1;
+  }
+
+  // mxconfig.driver   = HUB75_I2S_CFG::SHIFTREG;
+
+#if defined(ARDUINO_ADAFRUIT_MATRIXPORTAL_ESP32S3) // MatrixPortal ESP32-S3
```

Comment:
As mentioned in another comment, this should in fact be a runtime option.
I know it will/may be a lot of work/selection options, but I hate the idea that such configuration is compile-time based.

Reply:
I'm a bit surprised by this comment as I messaged you during the development and specifically discussed the challenges relating to the usability of configuration when so many pins are needed and that they are non configurable if you are using one of the ready made boards such as the Adafruit Matrix Portal S3

Reply:
I'm sorry if I forgot, I interact with several (possibly in range of 50) users and developers at any time.
So things may slip my mind and I do not heave the time to revisit all communication made. I apologise.

Nevertheless users may build clones or try their own PCBs following HW layout of the existing boards. So it would be more natural to select any Hub75 compatible board instead of compile-time choice. _Similar as it is with Ethernet_.

After all, the only difference I can see is in GPIO allocation.

Reply:
The ideal solution would be a total refactor whereby the front-end requests what config the bus requires, rather than the current mess of having a large number of elements that you then need to work out what to hide for that bus type. Then you can remove all the isType calls.

But this is of course a big change and should be a PR in it's own right

Reply:
I agree that it is a mess. Still, a little better than it was.